### PR TITLE
Integrate a feature flag in the `CustomViewsSelector` component

### PR DESCRIPTION
**Pull request by CarlosCortizasCT**
#### Summary

Integrate a feature flag in the `CustomViewsSelector` component

#### Description

In this PR we introduce a new feature flag usage so the `CustomViewsSelector` takes into account the feature flag we've already set up for this feature.

Without this, the component will always try to fetch the available Custom Views, even if the feature is disabled.

### CODE REVIEW

File: packages/application-components/src/components/custom-views/custom-views-selector/use-custom-views-connector.ts [hunk 1]
```diff
@@ -21,6 +23,10 @@ type TUseCustomViewsFetcher = (props: TUseCustomViewsFetcherParams) => {
 export const useCustomViewsConnector: TUseCustomViewsFetcher = ({
   customViewLocatorCode,
 }) => {
+  const areCustomViewsEnabled =
+    useFeatureToggle(featureFlags.CUSTOM_VIEWS) &&
+    process.env.DISABLE_CUSTOM_VIEWS_FEATURE !== 'true';
```
Comment (CarlosCortizasCT) [inline packages/application-components/src/components/custom-views/custom-views-selector/use-custom-views-connector.ts:28]: This is to easily be able to disable the feature when testing consumer applications so the request to fetch the Custom Views will not be sent.

File: packages/application-shell/src/components/application-shell/application-shell.spec.js [hunk 2]
```diff
@@ -109,6 +110,9 @@ const renderApp = (ui, options = {}) => {
     ...(renderNodeAsChildren
       ? { children: jsxElem }
       : { render: () => jsxElem }),
+    defaultFeatureFlags: {
```
Comment (CarlosCortizasCT) [inline packages/application-shell/src/components/application-shell/application-shell.spec.js:113]: Mock Custom Views feature flag so it's enabled in the tests.

**Review by emmenko — APPROVED** [review]
Very nice! 🙌

**Comment by emmenko** [comment]
@CarlosCortizasCT Don't forget the changeset 😉

**Review by chloe0592 — APPROVED** [review]
Thank you @CarlosCortizasCT! Looks good to me! 🙌 
Just don't forget the changeset 😄

File: packages/application-shell/src/components/application-shell/application-shell.spec.js [hunk 3]
```diff
@@ -109,6 +110,9 @@ const renderApp = (ui, options = {}) => {
...
+      [featureFlags.MAIN_NAVIGATION]: true,
```
[the remark this answers is not in the document]
Comment (CarlosCortizasCT) [reply packages/application-shell/src/components/application-shell/application-shell.spec.js]: Thanks Klaudija!

Actually, this one is not needed so I removed it here: d26cdd48301a01e5a14d6526c6e93f2dbe462708

**Pull request merged**